repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py

```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
     description = '''No description has been provided for the InplaceUpgrade workflow.'''
 
     class FactsCollectionPhase(Phase):
+        '''Operating System Version Detection, IP, Hostname, RPMs, Services,
+        Augeas.
+        '''
         name = 'Facts collection'
         filter = TagFilter(FactsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ChecksPhase(Phase):
+        '''Inhibitors - Blockers, Shared Objects, Binaries, Languages,
+        Packages, Load Balance Support, User Management, Version Control
+        Systems, FreeRADIUS, IPA Bind Dyndb LDAP, SSSD, NTP, OpenLDAP, YPSERV,
+        Network, Databases, SELinux, Storage, System, HTTPD, OpenSSH, Postfix,
+        SQUID, DoveCot, Bind9 Configuration, RSYSLOG.
+        '''
         name = 'Checks'
         filter = TagFilter(ChecksPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ReportsPhase(Phase):
+        '''Provide user with the result of the checks. Let user interactively
+        decide on multiple-choice solution for specific upgrade issues.
+        '''
         name = 'Reports'
         filter = TagFilter(ReportPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class AttachPackageReposPhase(Phase):
+        '''Make sure the repositories of the new system are available.'''
         name = 'AttachPackageRepos'
         filter = TagFilter(AttachPackageReposPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class PlanningPhase(Phase):
+        '''RPM Install transaction (feasibility checks).'''
         name = 'Planning'
         filter = TagFilter(PlanningPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class DownloadPhase(Phase):
+        '''Download of all things needed (RPMs, Kernel images, Anaconda image
+        (very big maybe), etc.)
+        '''
```

Comment:
here we do everything connected to the transaction - get target rpm repositories, check the rpm transaction, download all packages... 
From that point, we should rename the phase in future to fit better to what we do here.